core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/share/impl/EmailShareServiceImpl.java

```diff
@@ -135,14 +138,23 @@ public final void share(final SlingHttpServletRequest request, final SlingHttpSe
 
     private final void share(final Config config, final ValueMap shareParameters, final String emailTemplatePath) throws ShareException {
         final String[] emailAddresses = StringUtils.split(shareParameters.get(EMAIL_ADDRESSES, ""), ",");
-        final String[] assetPaths = shareParameters.get(ASSET_PATHS, String[].class);
+        final String[] assetEncodedPaths = shareParameters.get(ASSET_PATHS, String[].class);
 
         // Check to ensure the minimum set of e-mail parameters are provided; Throw exception if not.
         if (emailAddresses == null || emailAddresses.length == 0) {
             throw new ShareException("At least one e-mail address is required to share");
-        } else if (assetPaths == null || assetPaths.length == 0) {
+        } else if (assetEncodedPaths == null || assetEncodedPaths.length == 0) {
             throw new ShareException("At least one asset is required to share");
         }
+        int arrayLength = assetEncodedPaths.length;
```

Comment:
This looks rather .. verbose? Can't we do a nice stream() online 141... i think its reasonable to catch encoding exceptions and use the string as-is?

```
final String[] assetPaths = Stream.of(shareParameters.get(ASSET_PATHS, String[]{}))
.map(p -> try { return URLDecoder.decode(p, "UTF-8"); } 
                  catch (UnsupportedEncodingException e)  { log.warn("Could not decode path [ {} ], using as is", p); return p; })
.filter(p -> StringUtils::isBlank)
.toArray();
```
or something?